cachito/workers/pkg_managers/yarn.py

```diff
@@ -69,7 +70,37 @@ def get_yarn_proxy_repo_username(request_id):
     return f"cachito-yarn-{request_id}"
 
 
-def _get_deps(yarn_lock, file_deps_allowlist):
+def find_reachable_deps(yarn_lock, dep):
+    """
+    Get a set of all dependencies reachable by a top-level dependency using BFS.
+
+    :param yarn_lock: yarn.lock file as a dictionary
+    :param dep: top-level dependency in package.json
+    :return: set of all dependencies reachable by dep
+    :rtype: set
+    """
+    reachable_deps = set()
+    bfs_queue = deque()
+    reachable_deps.add(dep)
+    bfs_queue.append(dep)
+    while bfs_queue:
+        current_dep = bfs_queue.popleft()
+        for dep_identifier, dep_data in yarn_lock.items():
```

Comment:
How dep_identifier looks here?

Reply:
Anything from `istanbul-reports@^1.0.0-alpha` to `align-text@^0.1.1, align-text@^0.1.3`, this is why I use the for-loop here to iterate through `yarn.lock` and check if `current_dep` is included in `dep_identifier`.

e.g.
```
current_dep = "align-text@^0.1.3"
dep_identifier = "align-text@^0.1.1, align-text@^0.1.3"
```

Reply:
so `dep_identifier` is a list/tuple then?

Reply:
It's a tuple.